plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6

```diff
@@ -415,6 +422,210 @@ createWidget("discourse-poll-info", {
   }
 });
 
+function fetchGroupedVoters(data) {
+  return ajax("/polls/grouped_poll_results.json", { data }).catch(error => {
+    if (error) {
+      popupAjaxError(error);
+    } else {
+      bootbox.alert(I18n.t("poll.error_while_fetching_voters"));
+    }
+  });
+}
+
+function transformUserFieldToLabel(fieldName) {
+  let newName = fieldName;
+  return newName.replace(/_/g, " ");
+}
+
+createWidget("discourse-poll-grouped-pies", {
+  tagName: "div.poll-grouped-pies",
+  buildAttributes(attrs) {
+    return {
+      id: `poll-results-grouped-pie-charts-${attrs.id}`
+    };
+  },
+
+  html(attrs) {
+    const fields = Object.assign({}, attrs.groupableUserFields);
+    later(() => {
+      attrs.groupedBy = attrs.groupedBy || fields[0];
+      const parent = document.getElementById(
+        `poll-results-grouped-pie-charts-${attrs.id}`
+      );
+      const fieldSelectId = `field-select-${attrs.id}`;
+
+      const existingSel = document.getElementById(fieldSelectId);
+      if (existingSel) {
+        existingSel.parentNode.removeChild(existingSel);
+      }
+
+      const sel = document.createElement("select");
```

Comment:
`groupBySelect` would probably be a better name, you use `sel` multiple lines later, and without context we don't know what it is